common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1568,7 +1571,8 @@ def ajax_url(self):
         return self.handler_url(self.xmodule_instance, 'xmodule_handler', '', '').rstrip('/?')
 
     def get_block(self, block_id):
-        return self.get_module(self.descriptor_runtime.get_block(block_id))
+        block = self.descriptor_runtime.get_block(block_id)
+        return self.get_module(block) if block else None
```

Comment:
@cpennington When `get_children` is called after `_child_instances` has been deleted in `bind_for_student`, sometimes `get_block` returns None. When this happens `get_module` causes the following error. I am not sure why this doesn't happen on the first call to `get_children`. But I have set this to return `None` immediately if the block is `None`. It this the right way to do this? 

```
TypeError: Unknown object type in has_access(): '<type 'NoneType'>'
2015-02-09 06:55:19,854 ERROR 5863 [django.request] base.py:213 - Internal Server Error: /courses/edX/DemoX/Demo_Course/courseware/d8a6192ade314473a78242dfeedfbf5b/edx_introduction/
Traceback (most recent call last):
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 109, in get_response
    response = callback(request, *callback_args, **callback_kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/contrib/auth/decorators.py", line 20, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/utils/decorators.py", line 91, in _wrapped_view
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/venvs/edxapp/local/lib/python2.7/site-packages/django/views/decorators/cache.py", line 75, in _cache_controlled
    response = viewfunc(request, *args, **kw)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/views.py", line 40, in inner
    response = view_func(request, *args, **kwargs)
  File "/edx/app/edxapp/edx-platform/common/djangoapps/util/db.py", line 42, in wrapper
    return func(*args, **kwargs)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/views.py", line 336, in index
    return _index_bulk_op(request, course_key, chapter, section, position)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/views.py", line 386, in _index_bulk_op
    'accordion': render_accordion(request, course, chapter, section, field_data_cache),
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/views.py", line 131, in render_accordion
    toc = toc_for_course(request, course, chapter, section, field_data_cache)
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 160, in toc_for_course
    for chapter in course_module.get_display_items():
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 447, in get_display_items
    for child in self.get_children():
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 663, in get_children
    child_descriptors = self.get_child_descriptors()
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 691, in get_child_descriptors
    return self.descriptor.get_children()
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 416, in get_children
    child = self.runtime.get_block(child_loc)
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1571, in get_block
    return self.get_module(self.descriptor_runtime.get_block(block_id))
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 382, in inner_get_module
    request_token=request_token,
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/module_render.py", line 694, in get_module_for_descriptor_internal
    if not has_access(user, 'load', descriptor, course_id):
  File "/edx/app/edxapp/edx-platform/lms/djangoapps/courseware/access.py", line 104, in has_access
    .format(type(obj)))
TypeError: Unknown object type in has_access(): '<type 'NoneType'>'
```